gufe/chemicalsystem.py

```diff
@@ -29,6 +32,8 @@ class ChemicalSystem(Serializable, abc.Mapping):
 
     """
 
+    _storage_path = "setup/systems/{md5}.json"
```

Comment:
The name here is misleading. It should be called `_label_template`. Regardless of what storage we're using, we have a universal label for these objects. This is the same idea as the labels we use for data, which must be known by the `Protocol` and its `Results` object.

To use the storage we've been setting up so far, we need these labels. My thoughts here were:

* it would be nice if those labels were at least somewhat informative
* it would be nice if those labels carried file extensions so people could figure out what to expect

My approach was to create an implicit structure like:

```
setup/
  components/
    ASDF.sdf
    FGHJ.json
    HJKL.sdf
  settings/
    ZXCV.json
  systems/
    QWER.json
    VBNM.json
  transformations/
    UIOP.json
```

The alternative is just have the label be the MD5 representation -- no namespace, no extension:

```
ASDF
FGHJ
HJKL
QWER
UIOP
VBNM
ZXCV
```

From the computer's standpoint, this makes no difference. From the user's standpoint, I think having the structure is more clear. Those labels are what are written into the storage as references to objects; how they are interpreted as file paths or whatnot is up to the storage backend. I think that having some sense of the structure may be particularly useful when debugging.

To have that structure we need a mapping of class to something like those strings: specifically something which provides the namespace where the objects should live and the file extension. If that mapping is outside the class itself, we hit all the issues I listed in section 2 of the [long comment above](https://github.com/OpenFreeEnergy/gufe/pull/30#issuecomment-1172045531). 

In particular, I think the concerns from @dotsdl and @richardjgowers are about the possibility of swapping out for an external storage stack. But as I discuss in sections 2.4 and 2.6 of that comment, using a consistent label for these makes it easier to use an external storage stack.

Making that mapping of class to label template be part of storage means that each storage implementation must be aware of every contributed package (or each contributed package must be aware of every storage stack). Additionally, as soon as one storage package uses a different labeling scheme, we lose some potentially interesting functionality that is available if all storage systems agree on the label. Furthermore, it might be necessary for that storage to maintain a mapping of MD5 hash to its internally-defined label, e.g.,:

```
{ "ASDF": "setup/components/ASDF.sdf",
  "FGHI": "setup/components/FGHI.json",
   ...
}
```

These issues are avoided by simple including this class variable, (which, in most cases, can probably be set in a superclass, so that contributors don't even need to know about it).

---

**tl;dr: I think we have 3 options:**

1. What I'm doing. Make a user-friendly label part of the object.
2. Drop user friendly labels. Sacrifice the user-friendliness of the labels, but this keeps the idea of label out of the objects.
3. Move the user-friendly label to storage. This isn't impossible, but it will make life harder for contributors and for us. I think this is the worst option.